src/main/kotlin/com/team4099/robot2023/subsystems/elevator/ElevatorIOSim.kt

```diff
@@ -0,0 +1,146 @@
+                        elevatorSim,
+                        "Extension Motor",
+                    )
+                ),
+                baseCurrentLimit = 60.amps,
+                firstStageTemperatureLimit = 10.celsius,
+                firstStageCurrentLimit = 45.amps,
+                motorShutDownThreshold = 20.celsius
+            )
+        )
+    }
+
+    private val elevatorController =
+        PIDController(ElevatorConstants.SIM_KP, ElevatorConstants.SIM_KI, ElevatorConstants.SIM_KD)
+
+    private var lastAppliedVoltage = 0.0.volts
+
+    override fun updateInputs(inputs: ElevatorIO.ElevatorInputs) {
+        elevatorSim.update(Constants.Universal.LOOP_PERIOD_TIME.inSeconds)
+
+        inputs.elevatorPosition = elevatorSim.positionMeters.meters
+        inputs.elevatorVelocity = elevatorSim.velocityMetersPerSecond.meters.perSecond
+
+        inputs.leaderTempCelcius = 0.0.celsius
+        inputs.leaderStatorCurrent = 0.0.amps
+        inputs.leaderSupplyCurrent = elevatorSim.currentDrawAmps.amps / 2
+        inputs.leaderAppliedVoltage = lastAppliedVoltage
+
+        inputs.followerTempCelcius = 0.0.celsius
+        inputs.followerStatorCurrent = 0.0.amps
+        inputs.followerSupplyCurrent = elevatorSim.currentDrawAmps.amps / 2
+        inputs.followerAppliedVoltage = lastAppliedVoltage
+
+        inputs.leaderRawPosition = 0.0
+        inputs.followerRawPosition = 0.0
+
+        inputs.isSimulating = true
+
+        RoboRioSim.setVInVoltage(
+            BatterySim.calculateDefaultBatteryLoadedVoltage(elevatorSim.currentDrawAmps)
+        )
+    }
+
+    /**
+     * Sets the voltage of the elevator motors but also checks to make sure elevator doesn't exceed
+     * limit
+     *
+     * @param voltage the voltage to set the motor to
+     */
+    override fun setOutputVoltage(voltage: ElectricalPotential) {
+        val clampedVoltage =
+            clamp(
+                voltage,
+                -ElevatorConstants.VOLTAGE_COMPENSATION,
+                ElevatorConstants.VOLTAGE_COMPENSATION
+            )
+
+        lastAppliedVoltage = clampedVoltage
+
+        elevatorSim.setInputVoltage(clampedVoltage.inVolts)
+    }
+
+    /**
+     * Sets the voltage of the elevator motors but also checks to make sure elevator doesn't exceed
+     * limit
+     *
+     * @param position the target position the PID controller will use
+     * @param feedforward change in voltage to account for external forces on the system
+     */
+    override fun setPosition(position: Length, feedforward: ElectricalPotential) {
+        val ff =
+            clamp(
+                feedforward,
+                -ElevatorConstants.VOLTAGE_COMPENSATION,
+                ElevatorConstants.VOLTAGE_COMPENSATION
+            )
+        val feedback = elevatorController.calculate(elevatorSim.positionMeters.meters, position)
+
+        lastAppliedVoltage = ff + feedback
+        elevatorSim.setInputVoltage((ff + feedback).inVolts)
```

Comment:
should use the `setOutputVoltage` you created so you don't have to worry about the `lastAppliedVoltage` setting